### PR TITLE
Replace calls to malloc and strdup with wrapper functions that check the return values

### PR DESCRIPTION
**Pull request by lokedhs**

### CODE REVIEW

File: cartridge.c [hunk 1]
```diff
@@ -52,7 +52,7 @@ void cartridge_init(char *filename)
   FILE *fp;
   int file_size = 0;
 
-  memory = (BYTE *)malloc(sizeof(BYTE) * CARTRIDGESIZE);
+  memory = (BYTE *)xmalloc(sizeof(BYTE) * CARTRIDGESIZE);
```
Comment (larsbrinkhoff) [inline cartridge.c]: Could you please remove these unnecessary casts here and elsewhere?

File: cpu.c [hunk 2]
```diff
@@ -441,7 +441,7 @@ void cpu_init()
 {
   int i;
 
-  cpu = (struct cpu *)malloc(sizeof(struct cpu));
+  cpu = (struct cpu *)xmalloc(sizeof(struct cpu));
   if(!cpu) {
```
Comment (larsbrinkhoff) [inline cpu.c]: This already checks for `NULL` from `malloc`.  Either remove the check, or use the original code.

File: floppy_msa.c [hunk 3]
```diff
@@ -64,7 +64,7 @@ static void load_file(struct floppy *fl, FILE *fp)
 
   track_size = fl->sectors * SECSIZE;
   
-  track_data = malloc(track_size);
+  track_data = xmalloc(track_size);
   if(track_data == NULL) {
     ERROR("Unable to allocate track space");
```
Comment (larsbrinkhoff) [inline floppy_msa.c]: This is supposedly an error which is ok to display and then continue execution.  Your version aborts immediately.

File: ikbd.c [hunk 4]
```diff
@@ -278,7 +278,7 @@ static int ikbd_state_collect(struct mmu_state *state)
    */
 
   state->size = 20+IKBDFIFO+sizeof ikbd_cmdbuf+sizeof(void (*)(void));
-  state->data = (char *)malloc(state->size);
+  state->data = (char *)xmalloc(state->size);
   if(state->data == NULL)
     return STATE_INVALID;
```
Comment (larsbrinkhoff) [inline ikbd.c]: Error supposedly handled elsewhere.

File: main.c [hunk 5]
```diff
@@ -239,3 +239,23 @@ int main(int argc, char *argv[])
   while(cpu_run(CPU_RUN));
   return 0;
 }
+
+void *xmalloc(size_t size)
+{
+  void *result = malloc(size);
+  if(result == NULL) {
+    fprintf(stderr, "Unable to allocate %lld bytes", (long long)size);
+    exit(1);
```
Comment (larsbrinkhoff) [inline main.c]: Please try to use `FATAL` from `diag.h`.

File: prefs.c [hunk 6]
```diff
@@ -6,7 +6,7 @@ struct prefs prefs;
 
 void prefs_default()
 {
-  prefs.tosimage = strdup("tos.img");
+  prefs.tosimage = xstrdup("tos.img");
```
Comment (larsbrinkhoff) [inline prefs.c]: It's not necessary to `strdup` a constant string.

File: main.c (hunk 5 above)
Comment (lokedhs) [inline main.c]: FATAL does not do what you'd expect. First of all, it required a struct mmu
parameter. Secondly it doesn't exit.

Comment (larsbrinkhoff) [inline main.c]: Ok, I can see this may be hard to understand.  I'll fix it for you later.

**Comment by larsbrinkhoff** [comment]
It's a matter of taste whether to squash all these commits together to avoid clutter in the history.  My preference would be to do that in cases like these when follow-up commits just correct mistakes in the first commit.

File: prefs.c (hunk 6 above)
Comment (stefanberndtsson) [inline prefs.c]: Yep, agreed.

**Comment by stefanberndtsson** [comment]
In general, I agree with the squashing bit, but since I do the clutter stuff myself all the time, I won't care about it here, as long as the commits actually connect with the issue trying to be solved.

Remove the tos.img thing, and I'll merge it.

**Comment by larsbrinkhoff** [comment]
@lokedhs Don't forget to fix your `master` branch.

**Pull request merged**